### ISSUE TITLE
Replace scheduler-specific command-line options

### ISSUE THREAD

**Issue by vkarak** [eth-cscs/reframe#486]
The `--partition`, `--account`, `--nodelist` etc. options are actually scheduler-specific and, in my opinion, they should not be part of the command-line interface. Instead, the should be passed through a more generic mechanism such as the `--job-option` switch. There are advantages and disadvantages to what I suggest here:

Pros:
- Cleaner command-line
- Less implementation overhead for scheduler backends, since the user is responsible for passing the correct backend options in the special case that he wants to modify how an access to a specific partition is gained. A backend implementor need not be an expert of the scheduler.
- More uniform behaviour for the scheduler backends. For example, now we assume, at the supposedly scheduler-agnostic frontend, a SLURM-specific syntax for the node list.
- Avoid subtle bugs such as the one described in #75.

Cons:
- We lose the automatic mapping of the command-line options to the different backends. This is only true for the `--partition` one. The rest are simply ignored by all other schedulers except Slurm.
- Perhaps more typing

**Comment by victorusu** [comment]
@vkarak, I think the overhead of typing more in the command line exists only when testing... Most of the ReFrame execution that we have seen is performed by robots...

**Comment by vkarak** [comment]
The problem is conceptual. You can't have command line options that essentially map to specific backend. This is wrong. You should still be able to pass these options, but in a more generic way.

**Comment by vkarak** [comment]
The fact that these options are there is pure legacy. They are coming from the pre-ReFrame regression framework.

**Comment by victorusu** [comment]
@vkarak, I completely agree with you!

**Comment by vkarak** [comment]
We just had a chat with @rsarm and we settled on a nice solution:

- All the scheduler-specific options with will be deprecated.
- There will only be left the `--job-option`, which will gain a short option variant, `-J`.
- The general syntax of this option will be in the form of `-J key=value`, e.g., `-J partition=debug`, `-J q=debug` etc. The frontend will translate `key` to a scheduler option according to the following rules:
    - If `key` starts with a `-` or with `#`, `-J`'s argument will be passed as-is to the backend scheduler. This allows to pass any non-standard options plus extra preamble directives.
    - If `len(key)` is 1 then `-key value` will be passed to the backend scheduler.
    - If `len(key)` is greater than 1, then `--key=value` will be passed to the backend scheduler.

In order to avoid the refactoring of the internal `sched_xxx` variables, we decided it to leave them for the moment, in order to be able to ship this feature in 3.0. We will later remove all these `sched_xxx` variables internal and leave only the `sched_options`.
